pkg/chargeback/http_test.go

```diff
@@ -616,3 +616,317 @@ func TestAPIV2ReportsGet(t *testing.T) {
 		})
 	}
 }
+
+//for v2 endpoints TableHidden
+func apiReportV2URLTable(reportName string) string {
+	return path.Join(APIV2Reports, reportName, "table")
+}
+
+func TestAPIV2ReportsTable(t *testing.T) {
+	const namespace = "default"
+	const testReportName = "test-report"
+	const testQueryName = "test-query"
+	reportStart := time.Time{}
+	reportEnd := reportStart.AddDate(0, 1, 0)
+
+	tests := map[string]struct {
+		reportStatus v1alpha1.ReportStatus
+		reportName   string
+
+		report      *v1alpha1.Report
+		query       *v1alpha1.ReportGenerationQuery
+		prestoTable *v1alpha1.PrestoTable
+
+		expectedStatusCode int
+		expectedAPIError   string
+
+		queryerPrepareFunc func(*mockpresto.MockExecQueryer) []presto.Row
+		expectedResults    *GetReportResults
+	}{
+		"report-finished-with-results-tableHidden-false": {
+			reportName: testReportName,
+			report:     newTestReport(testReportName, namespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{Phase: v1alpha1.ReportPhaseFinished}),
+			query: newTestReportGenQuery(testQueryName, namespace, []v1alpha1.ReportGenerationQueryColumn{
+				{
+					Name:        "timestamp",
+					Type:        "timestamp",
+					TableHidden: false,
+				},
+				{
+					Name:        "foo",
+					Type:        "double",
+					TableHidden: false,
+				},
+			},
+			),
+			prestoTable: newTestPrestoTable(testReportName, namespace, []hive.Column{
+				{
+					Name: "timestamp",
+					Type: "timestamp",
+				},
+				{
+					Name: "foo",
+					Type: "double",
+				},
+			},
+			),
+			queryerPrepareFunc: func(mock *mockpresto.MockExecQueryer) []presto.Row {
+				result := []presto.Row{
+					{
+						"timestamp": time.Time{},
+						"foo":       1,
+					},
+				}
+				mock.EXPECT().Query(gomock.Any()).Return(result, nil)
+				return result
+			},
+			expectedStatusCode: http.StatusOK,
+			expectedResults: &GetReportResults{
```

Comment:
These expectedResults aren't correct. The timestamp value is missing.

Reply:
fixed by bool change